File: fix_categories_script.js
const fs = require('fs');

function getCategoryQuartos(name) {
    name = name.toLowerCase();
    if (name.includes('mesa') && name.includes('cabeceira')) return 'mesasCabeceira';
    if (name.includes('cabeceira')) return 'cabeceiras';
    if (name.includes('cama')) return 'camas';
    if (name.includes('cómoda') || name.includes('comoda')) return 'comodas';
    if (name.includes('camiseiro')) return 'camiseiros';
    if (name.includes('sommier')) return 'sommiers';
    return 'all';
}

function getCategorySalas(name) {
    name = name.toLowerCase();
    if (name.includes('sofá') || name.includes('sofa')) return 'sofas';
    if (name.includes('cadeirão') || name.includes('cadeirao') || name.includes('poltrona')) return 'cadeiroes';
    if (name.includes('mesa') || name.includes('cadeira') || name.includes('banqueta') || name.includes('banco')) return 'mesasEcadeiras';
    if (name.includes('móvel') || name.includes('movel') || name.includes('aparador') || name.includes('estante') || name.includes('vitrine') || name.includes('sapateira') || name.includes('secretária') || name.includes('secretaria') || name.includes('base tv') || name.includes('prateleira')) return 'moveis';
    return 'all';
}

function getCategoryColchoes(name) {
    name = name.toLowerCase();
    if (name.includes('colchão') || name.includes('colchao')) return 'colchoes';
    return 'complementos';
}

function fixFile(filepath, getCatFunc) {
    if (!fs.existsSync(filepath)) return;
    let content = fs.readFileSync(filepath, 'utf-8');
    
    let result = content.replace(/<a([^>]*class="[^"]*product[^"]*"[^>]*)>/g, (match, p1) => {
        // Find the following h3 within the product block, roughly
        // Since we are doing regex on html, we need to locate the name.
        return match;
    });
    // Regular expression replacing isn't safe for HTML with nested tags
}

fixFile('quartos.html', getCategoryQuartos);
fixFile('salas.html', getCategorySalas);
fixFile('colchoes.html', getCategoryColchoes);

File: fix_categories_script.py
import os
import bs4

def get_category_quartos(name):
    name = name.lower()
    if 'mesa' in name and 'cabeceira' in name:
        return 'mesasCabeceira'
    if 'cabeceira' in name:
        return 'cabeceiras'
    if 'cama' in name:
        return 'camas'
    if 'cómoda' in name or 'comoda' in name:
        return 'comodas'
    if 'camiseiro' in name:
        return 'camiseiros'
    if 'sommier' in name:
        return 'sommiers'
    return 'all'

def get_category_salas(name):
    name = name.lower()
    if 'sofá' in name or 'sofa' in name:
        return 'sofas'
    if 'cadeirão' in name or 'cadeirao' in name or 'poltrona' in name:
        return 'cadeiroes'
    if 'mesa' in name or 'cadeira' in name or 'banqueta' in name or 'banco' in name:
        return 'mesasEcadeiras'
    if 'móvel' in name or 'movel' in name or 'aparador' in name or 'estante' in name or 'vitrine' in name or 'sapateira' in name or 'secretária' in name or 'secretaria' in name or 'base tv' in name or 'prateleira' in name:
        return 'moveis'
    return 'all'

def get_category_colchoes(name):
    name = name.lower()
    if 'colchão' in name or 'colchao' in name:
        return 'colchoes'
    return 'complementos'

def fix_file(filepath, get_category_func):
    if not os.path.exists(filepath):
        return
    with open(filepath, 'r', encoding='utf-8') as f:
        soup = bs4.BeautifulSoup(f.read(), 'html.parser')
    
    products = soup.find_all('a', class_=lambda c: c and 'product' in c)
    for p in products:
        h3 = p.find('h3')
        if not h3:
            continue
        name = h3.text.strip()
        cat = get_category_func(name)
        p['data-category'] = cat
        
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(str(soup))
    print(f"Fixed {filepath}")

fix_file('quartos.html', get_category_quartos)
fix_file('salas.html', get_category_salas)
fix_file('colchoes.html', get_category_colchoes)
